src/Icons/Models/IconLink.cs

```diff
@@ -0,0 +1,212 @@
+﻿#nullable enable
+
+using System.Text;
+using AngleSharp.Dom;
+using Bit.Icons.Extensions;
+using Bit.Icons.Services;
+
+namespace Bit.Icons.Models;
+
+public class IconLink
+{
+    private static readonly HashSet<string> _iconRels = new() { "icon", "apple-touch-icon", "shortcut icon" };
+    private static readonly HashSet<string> _blocklistedRels = new() { "preload", "image_src", "preconnect", "canonical", "alternate", "stylesheet" };
+    private static readonly HashSet<string> _iconExtensions = new() { ".ico", ".png", ".jpg", ".jpeg" };
```

Comment:
If we pass `StringComparer.InvariantCultureIgnoreCase` into the constructor we don't have to `.ToLower()` the inputs into the `.Contains` calls and save a few string allocations.

Reply:
good catch! I didn't think about it at all, just copied 😳